tests/unit/CMakeLists.txt

```diff
@@ -17,6 +21,11 @@ if(BUILD_TESTING) #set by Ctest. Also set in the integration build environment.
         endif()
     endif()
 
+    add_compile_options("-fpermissive")
+    add_definitions(
```

Comment:
`add_definitions()` has been superseded by `add_compile_definitions()` for preprocessor definitions (which this doesn't seem to be) and `add_compile_options()` for other flags. I propose that you use `add_compile_options()` for this as well.